es_schema.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass, field
+import typing
+from typing import Annotated
+
+from arcaflow_plugin_sdk import validation, schema
+
+
+@dataclass
+class ElasticsearchStorage:
+    username_envvar: Annotated[str] = field(
+        metadata={
+            'name': 'username_envvar',
+            'description': """Environment variable that points to """
+            """an authorized username for the given Elasticsearch instance."""
+        }
+    )
+    username: Annotated[typing.Optional[str]] = field(
+        metadata={
+            'name': 'username',
+            'description': """Authorized username for the given """
+            """Elasticsearch instance."""
+        }
+    )
+    password_envvar: Annotated[str, validation.min(0)] = field(
+        metadata={
+            'name': 'password_envvar',
+            'description': """Environment variable that points to the""" 
+            """password for the given username."""
+        }
+    )
+    password: Annotated[typing.Optional[str], validation.min(0)] = field(
+        metadata={
+            'name': 'password',
+            'description': """Password for username for the given """
+            """username."""
+        }
+    )
+    url_envvar: Annotated[str] = field(
+        metadata={
+            'name': 'url',
+            'description': """Environment variable that points """
+            """to the URL of your Elasticsearch instance."""
+        }
+    )
+    url: Annotated[typing.Optional[str]] = field(
+        metadata={
+            'name': 'url',
+            'description': """URL to your Elasticsearch """
+            """instance."""
+        }
+    )
+    index: Annotated[str, validation.min(1)] = field(
+        metadata={
+            'name': 'index',
+            'description': """Name of the index that """
+            """will receive this data."""
+        }
+    )
+    data: Annotated[typing.Optional[schema.ANY_TYPE]] = field(
+        metadata={
+            'name': 'data',
```

Comment:
Lots of questions come up with this plug-in.

Typically one does not just index one document into Elasticsearch.  You have a series of N documents that you want indexed.  So while this is fine for an initial step, we need to discuss the nature of this as a plugin.

FWIW, we wrote [pyesbulk](https://pypi.org/project/pyesbulk/) to enable an efficient way of indexing a stream of data into Elasticsearch, with error handling, and able to handle multiple documents.

It is primarily used by the [pbench server code](https://github.com/distributed-system-analysis/pbench/blob/665132a898e068a6c488610ccc447fd33db68035/lib/pbench/server/indexer.py#L180).

Reply:
One document at a time isn't going to be worth it in a workflow, you'll either have a huge scaling problem trying to describe all the individual documents in the workflow, or you'll have too large a document to be useful in Elasticsearch.